crates/pipeline_manager/src/compiler.rs

```diff
@@ -123,8 +123,9 @@ impl ProgramStatus {
 /// Program configuration.
 #[derive(Clone, Debug, Eq, PartialEq, Default, Serialize, Deserialize, ToSchema)]
 pub struct ProgramConfig {
-    /// Request a compilation profile.
-    pub profile: CompilationProfile,
+    /// Compilation profile.
+    /// If none is specified, the compiler default compilation profile is used.
+    pub profile: Option<CompilationProfile>,
```

Comment:
any downside to adding `#[serde(flatten)]` here, so the user doesn't have to deal with an extra level of nesting in json?

Reply:
I assume you refer to flattening the config field in the program? I would lead to backwards incompatibility of existing API, though could be handled for already stored ones (as it is its own field in the database schema).
(CompilationProfile is just an enumeration, as such flattening would not have an effect I think). As this change is not directly related to the PR, I'd say to leave it to a later PR.

Reply:
Program configurations are now flattened, such that `profile` is a top-level property.